From a8e8e8fd5d59663a0472676ce40b66c4143b9f34 Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Mon, 4 Feb 2013 16:35:15 +0100
Subject: [PATCH] xbps_remove_pkg: ignore ELOOP in realpath() when checking
 symlinks.

Probably it is a broken symlink, so we don't have to care about it.
---
 lib/package_remove.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/package_remove.c b/lib/package_remove.c
index 4e69e82..78ddd7b 100644
--- lib/package_remove.c
+++ lib/package_remove.c
@@ -170,7 +170,7 @@ xbps_remove_pkg_files(struct xbps_handle *xhp,
 			 * point, so we will only remove dangling symlinks.
 			 */
 			if (realpath(path, buf) == NULL) {
-				if (errno != ENOENT) {
+				if (errno != ENOENT && errno != ELOOP) {
 					free(path);
 					rv = errno;
 					break;
-- 
1.8.1.1

